### DISCUSSION TITLE
TimeoutException after 0:00:20.000000: Future not completed

### DISCUSSION THREAD

**Discussion by letscodetpl** [serverpod/serverpod#2371] [Q&A] [answered]
Hi everyone, 
I have an issue that it throws TimeoutException after 0:00:20.000000: Future not completed. I am not really sure what to do next. Thanks for your help

What I am trying to do is insert a new row into database.

below is my code:

my visitor class

class: Visitor
table: visitor
fields:
  visitorName: String
  visitorRepresenting: String?
  whomVisiting: String?
  dateTimeIn: DateTime?
  dateTimeOut: DateTime?

My Endpoint:

class VisitorsEndpoint extends Endpoint {
  //New visitor added to DB
  Future<void> addNewVisitor(Session session, Visitor visitor) async {
    await Visitor.db.insertRow(session, visitor);
  }

  //Read visitor table from DB
  Future<List<Visitor>> getAllVisitors(Session session) async {
    return await Visitor.db.find(
      session,
      orderBy: (t) => t.id,
    );
  }

  //Update visitor to the DB
  Future<void> updateVisitor(Session session, Visitor visitor) async {
    await Visitor.db.updateRow(session, visitor);
  }

  //Delete Visitor from the DB
  Future<void> deleteVisitor(Session session, Visitor visitor) async {
    await Visitor.db.deleteRow(session, visitor);
  }
}

My UI
return Scaffold(
      body: Padding(
        padding: const EdgeInsets.only(top: 50.0, left: 150.0, right: 150.0),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.start,
          children: [
            const Text(
              'V I S I T O R  D E T A I L S',
              style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
            ),
            const SizedBox(height: 25),
            //Visitor name field
            TextField(
              controller: visitorNameEdittingController,
              decoration: const InputDecoration(
                hintText: "Visitor's name",
                border: OutlineInputBorder(),
              ),
            ),
            const SizedBox(height: 10),
            //Visitor representing field
            TextField(
              controller: visitorRepresentingEdittingController,
              decoration: const InputDecoration(
                hintText: "Visitor's representing",
                border: OutlineInputBorder(),
              ),
            ),
            const SizedBox(height: 10),
            //Visitor - whom visiting
            TextField(
              controller: visitorWhomVisitingController,
              decoration: const InputDecoration(
                hintText: "Whom visiting",
                border: OutlineInputBorder(),
              ),
            ),
            const SizedBox(height: 20),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
              children: [
                //btn check in
                MyButton(
                  text: "Check - in",
                  //press check-in to add visitor to DB then return to previous page
                  onTap: () {
                    var visitor = Visitor(
                      visitorName: visitorNameEdittingController.text,
                      dateTimeIn: DateTime.now(),
                      dateTimeOut: DateTime.now(),
                      visitorRepresenting:
                          visitorRepresentingEdittingController.text,
                      whomVisiting: visitorWhomVisitingController.text,
                    );
                    addVisitor(visitor);
                    visitorNameEdittingController.clear();
                    visitorRepresentingEdittingController.clear();
                    visitorWhomVisitingController.clear();
                    Navigator.pop(context);
                    Navigator.pushNamed(context, "/visitor_page");
                  },
                  width: 100,
                  height: 50,
                ),
                MyButton(
                  //btn cancel
                  text: "Cancel",
                  onTap: () => Navigator.pushNamed(context, '/visitor_page'),
                  width: 100,
                  height: 50,
                ),
              ],
            )
            //btn cancel
          ],
        ),
      ),
    );
  }
}

**Comment by SandPod** [comment, marked as answer]
Hey @letscodetpl  👋 

A couple of initial questions, are you getting the exception on the server or on the flutter app?

By using three ` symbols followed by "dart" you can format your code blocks making them easer to read.

Could you format the code examples in different blocks so that they become easier to inspect?

Here is an example of a formatted code block:

```dart
void main() {
  var i = 3;
  print(i);
}
```

**Reply by letscodetpl** [reply]
Hey @SandPod,
This was a 'silly me' moment, I was running on Chorme then decided to run on android emulator. should have changed to 10.0.2.2 instead of localhost.

it's all good now. 

Thank you..
